packages/vite-environment-provider-node-vm/src/index.ts

```diff
@@ -156,20 +156,7 @@ async function createNodeVmDevEnvironment(
               },
             },
           },
-          {
-            runInlinedModule: async (context, transformed, id) => {
-              const codeDefinition = \`'use strict';async (\${Object.keys(context).join(
-                ',',
-              )})=>{{\`;
-              const code = \`\${codeDefinition}\${transformed}\n}}\`;
-              const fn = eval(code, id);
-              await fn(...Object.values(context));
-              Object.freeze(context.__vite_ssr_exports__);
-            },
-            async runExternalModule(filepath) {
-              return import(filepath);
-            },
-          },
+          new ESModulesEvaluator()
```

Comment:
Replaced the custom code that was used here with `ESModulesEvaluator` from Vite.